### PR TITLE
more flexibility for lock api

### PR DESCRIPTION
**Pull request by cmyui**
let flame review this one

### CODE REVIEW

File: objects/redisLock.py [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
-    def acquire(self) -> None:
-        while not self.try_acquire():
-            time.sleep(RETRY_DELAY)
+    def acquire(
+        self,
+        expiry: float = DEFAULT_LOCK_EXPIRY,
+        retry_delay: float = DEFAULT_RETRY_DELAY,
+    ) -> None:
+        while not self._try_acquire(expiry):
+            time.sleep(retry_delay)
```
Comment (cmyui) [inline objects/redisLock.py]: maybe we should have a timeout as well

File: objects/redisLock.py [hunk 2]
```diff
@@ -6,20 +6,24 @@
...
 
-    async def acquire(self) -> None:
-        while not await self.try_acquire():
-            time.sleep(RETRY_DELAY)
+    async def acquire(
+        self,
+        expiry: float = DEFAULT_LOCK_EXPIRY,
+        retry_delay: float = DEFAULT_RETRY_DELAY,
+    ) -> None:
+        while not await self.try_acquire(expiry):
```
Comment (cmyui) [inline objects/redisLock.py]:
```suggestion
        while not await self._try_acquire(expiry):
```

File: objects/redisLock.py [hunk 3]
```diff
@@ -6,20 +6,24 @@
...
+DEFAULT_RETRY_DELAY = 0.05  # in seconds
 
 
 class redisLock:
     def __init__(self, key: str) -> None:
         self.key = key
 
-    async def try_acquire(self) -> Optional[bool]:
-        return await glob.redis.set(self.key, "1", ex=LOCK_EXPIRY, nx=True)
+    async def try_acquire(self, expiry: float) -> Optional[bool]:
```
Comment (cmyui) [inline objects/redisLock.py]:
```suggestion
    async def _try_acquire(self, expiry: float) -> Optional[bool]:
```

**Comment by infernalfire72** [comment]
👍 lgbt

**Review by infernalfire72 — APPROVED** [review]

**Pull request merged**